### DISCUSSION TITLE
How debug sending e-mail using the SMTP method

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#498] [General]
Hi,
I'm having trouble setting up the SMTP method on EME.
I also use the Easy WP SMTP plugin. With this plugin I get all my emails sent by my host's SMTP server, including those from EME when I use the default setting "WordPress Email (default)".

But if I then set EME to the same parameters as Easy WP SMTP it doesn't work. Is there a conflict? If I check "Debug SMTP" I can't then find the log file?

**Comment by liedekef** [comment]
The 2 plugins won't interfere, the problem will be elsewhere (I'm guessing with the server certificate). Debug SMTP in EME doesn't generate a logflle, but shows the smtp trace when sending a test mail via the submenu "Email management", tab "test email"
Also, don't forget that the only real reason something doesn't get through is only logged in your mailserver logfiles, so check those too.

**Comment by f14m07** [comment]
I'm bothered because I don't understand. Can you look at these screenshots. The settings between the 2 plugins are the same. I've checked the login and password several times. But the trace always shows the same problem
[Settings EasyWP SMTP.pdf](https://github.com/liedekef/events-made-easy/files/14760163/Settings.EasyWP.SMTP.pdf)
[Settings EME.pdf](https://github.com/liedekef/events-made-easy/files/14760164/Settings.EME.pdf)
[trace debug EME.pdf](https://github.com/liedekef/events-made-easy/files/14760165/trace.debug.EME.pdf)

**Reply by liedekef** [reply]
based on your output the password seems wrong ("incorrect authentication data"). Make sure you have no spaces before/after the username and password

**Reply by f14m07** [reply]
Is it possible the stored password is not the good one?

**Reply by liedekef** [reply]
Do a html-inspect on that field to see the password value ...

**Reply by f14m07** [reply]
I checked the WP_options table. There is eme_smtp_password with the password in clear text but there I had some strange characters:
a capital letter at the beginning instead of the lower case and after a '&' appeared 'amp;' which is not in the password.

**Reply by f14m07** [reply]
I do a html-inspect after the correction in the database table and the password is correct. But the problem is always there

**Reply by liedekef** [reply]
That's very weird ... I'll check the html-encoding issue here too (but normally it is the wordpress Options API doing the storing, not EME. The capital is really weird (and I never had that issue here, becaus that would render any password faulty)

**Reply by f14m07** [reply]
As I use the letter & in the password maybe there is a ..

**Reply by liedekef** [reply]
You should not fiddle with the wordpress options table yourself. It can very well be that wordpress expects everything to be html-encoded there. So I suggest to enter the password yourself again in EME and check again in the wordpress options table. The capital is something that should not occur ...

**Reply by f14m07** [reply]
I did and the password was stored the database with the following  "amp;" just after "&" which is part of the password. For example the password is jgld&Hard145 and in the database I have jgld&amp;Hard145

**Reply by f14m07** [reply]
no more capital

**Reply by f14m07** [reply]
a new html-inspect show the good password. So I do not understand the trace debug

**Reply by liedekef** [reply]
I think I understand where the "amp;"-part is coming from, but give me some time to test, it"s dinner time here :-)

**Reply by liedekef** [reply]
This fix should help: https://github.com/liedekef/events-made-easy/commit/4f1d2fa5f445c1f4e09a97617877cee2112ee497
Best is probably to just download the new eme-options.php file and replace yours.

**Reply by liedekef** [reply]
If it works, you've helped fix something really annoying :-)

**Reply by f14m07** [reply]
It works perfectly. Thanks a lot Franky
